pkg/repo/repotest/server.go

```diff
@@ -79,8 +79,15 @@ func NewServer(docroot string) *Server {
 
 // Server is an implementation of a repository server for testing.
 type Server struct {
-	docroot string
-	srv     *httptest.Server
+	docroot    string
+	srv        *httptest.Server
+	middleware http.HandlerFunc
+}
+
+// WithMiddleware injects middleware in front of the server. This can be used to inject
+// additional functionality like layering in an authentication frontend.
+func (s *Server) WithMiddleware(middleware http.HandlerFunc) {
+	s.middleware = middleware
```

Comment:
If we are enabling middleware, should we handle chaining multiple middlewares or leave it up to the user to pass a `HandlerFunc` that already has all the middleware composed?

Reply:
I can look further into this. I started with the idea of allowing a way to append middleware to the chain but landed on this design as a first pass.

Reply:
Because it is a test server, I think it is ok. The question just came up as I was looking at the code